src/easyscience/Objects/variable/descriptor_number.py

```diff
@@ -198,7 +240,7 @@ def error(self, value: float) -> None:
         else:
             self._scalar.variance = None
 
-    def convert_unit(self, unit_str: str) -> None:
+    def _convert_unit(self, unit_str: str) -> None:
         """
         Convert the value from one unit system to another.
```

Comment:
Add a sentence motivating why both `_convert_unit` and `convert_unit` exists (to avoid cyclic deps).